### ISSUE TITLE
Nextcloud App crashes on activity tab

### ISSUE THREAD

**Issue by vikarandris** [nextcloud/android#5333]
### Steps to reproduce
1. open activity tab
2. download file, or refresh tab
3. nextcloud chrashed

### Expected behaviour
- Tell us what should happen

### Actual behaviour
- Tell us what happens

### Environment data
Android version: 9

Device model: Huaewi p10

Stock or customized system:EMUI 9.1.0

Nextcloud app version: 3.1.10

Nextcloud server version: 17.0.2
Activity app version:2.10.1
### Logs

Insert your Nextcloud log here
************ CAUSE OF ERROR ************

java.net.SocketTimeoutException: failed to connect to taborozok.ddns.net/80.98.88.136 (port 443) from /192.168.0.21 (port 33628) after 10000ms
at libcore.io.IoBridge.connectErrno(IoBridge.java:185)
at libcore.io.IoBridge.connect(IoBridge.java:129)
at java.net.PlainSocketImpl.socketConnect(PlainSocketImpl.java:137)
at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:391)
at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:231)
at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:213)
at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:436)
at java.net.Socket.connect(Socket.java:621)
at okhttp3.internal.platform.AndroidPlatform.connectSocket(AndroidPlatform.java:73)
at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:247)
at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:167)
at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:258)
at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:135)
at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:114)
at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:127)
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:257)
at okhttp3.RealCall.execute(RealCall.java:93)
at com.nextcloud.common.OkHttpMethodBase.execute(OkHttpMethodBase.kt:126)
at com.nextcloud.common.NextcloudClient.execute(NextcloudClient.kt:67)
at com.owncloud.android.lib.resources.activities.GetActivitiesRemoteOperation.run(GetActivitiesRemoteOperation.java:131)
at com.nextcloud.common.NextcloudClient.execute(NextcloudClient.kt:63)
at com.owncloud.android.ui.fragment.FileDetailActivitiesFragment.lambda$fetchAndSetData$5$FileDetailActivitiesFragment(FileDetailActivitiesFragment.java:331)
at com.owncloud.android.ui.fragment.-$$Lambda$FileDetailActivitiesFragment$N7POe6TZvoGcuV2qP-niPxY_7t4.run(Unknown Source:12)
at java.lang.Thread.run(Thread.java:784)

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30100090
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: HUAWEI
Device: HWVTR
Model: VTR-L29
Id: HUAWEIVTR-L29
Product: VTR-L29

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: 252C432

**Comment by ezaquarii** [comment]
@tobiasKaminsky There is couple of similar errors in the backlog.

The main issue with the code is that `NextcloudClient.execute()` interface does not declare any exceptions and the underlying implementation throws checked exceptions.

To fix this problem:
1. add exceptions to the `execute()` interface and handle them at the point of call
2. catch those exceptions internally and return appropriate `RemoteOperationResult` with error status.

I believe 2. is the proper way of doing it, but the fix should follow in the library project.

**Comment by tobiasKaminsky** [comment]
Yes, let us go with 2. as this is what we currently also do: in client respond on a RemoteOperationResult.

**Comment by 4q1** [comment]
When will this update get pushed out to the android store?  I have the same issue, among many more. It's still an issue now.

**Comment by tobiasKaminsky** [comment]
Can you  try out our dev version (you can install it in parallel) and give us feedback: 
https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#dev-release

**Comment by vikarandris** [comment]
I try nextcloud-dev-20200211.apk
After download a file I get the same exception:

************ CAUSE OF ERROR ************

java.net.SocketTimeoutException: failed to connect to taborozok.ddns.net/80.98.88.136 (port 443) from /10.251.6.77 (port 38918) after 10000ms
	at libcore.io.IoBridge.connectErrno(IoBridge.java:185)
	at libcore.io.IoBridge.connect(IoBridge.java:129)
	at java.net.PlainSocketImpl.socketConnect(PlainSocketImpl.java:137)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:391)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:231)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:213)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:436)
	at java.net.Socket.connect(Socket.java:621)
	at okhttp3.internal.platform.AndroidPlatform.connectSocket(AndroidPlatform.java:73)
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:247)
	at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:167)
	at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:258)
	at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:135)
	at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:114)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:127)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:257)
	at okhttp3.RealCall.execute(RealCall.java:93)
	at com.nextcloud.common.OkHttpMethodBase.execute(OkHttpMethodBase.kt:126)
	at com.nextcloud.common.NextcloudClient.execute(NextcloudClient.kt:67)
	at com.owncloud.android.lib.resources.activities.GetActivitiesRemoteOperation.run(GetActivitiesRemoteOperation.java:131)
	at com.nextcloud.common.NextcloudClient.execute(NextcloudClient.kt:63)
	at com.owncloud.android.ui.fragment.FileDetailActivitiesFragment.lambda$fetchAndSetData$5$FileDetailActivitiesFragment(FileDetailActivitiesFragment.java:331)
	at com.owncloud.android.ui.fragment.-$$Lambda$FileDetailActivitiesFragment$N7POe6TZvoGcuV2qP-niPxY_7t4.run(Unknown Source:12)
	at java.lang.Thread.run(Thread.java:784)

************ APP INFORMATION ************
ID: com.nextcloud.android.beta
Version: 20200211
Build flavor: versionDev

************ DEVICE INFORMATION ************
Brand: HUAWEI
Device: HWVTR
Model: VTR-L29
Id: HUAWEIVTR-L29
Product: VTR-L29

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: 252C432

**Comment by vikarandris** [comment]
With same version:
I go to the activity tab and I get this exception:
************ CAUSE OF ERROR ************

javax.net.ssl.SSLHandshakeException: Connection closed by peer
	at org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
	at org.conscrypt.NativeSsl.doHandshake(NativeSsl.java:386)
	at org.conscrypt.ConscryptFileDescriptorSocket.startHandshake(ConscryptFileDescriptorSocket.java:225)
	at okhttp3.internal.connection.RealConnection.connectTls(RealConnection.java:320)
	at okhttp3.internal.connection.RealConnection.establishProtocol(RealConnection.java:284)
	at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:169)

	at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:258)
	at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:135)
	at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:114)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:127)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:257)
	at okhttp3.RealCall.execute(RealCall.java:93)
	at com.nextcloud.common.OkHttpMethodBase.execute(OkHttpMethodBase.kt:126)
	at com.nextcloud.common.NextcloudClient.execute(NextcloudClient.kt:67)
	at com.owncloud.android.lib.resources.activities.GetActivitiesRemoteOperation.run(GetActivitiesRemoteOperation.java:131)
	at com.nextcloud.common.NextcloudClient.execute(NextcloudClient.kt:63)
	at com.owncloud.android.ui.fragment.FileDetailActivitiesFragment.lambda$fetchAndSetData$5$FileDetailActivitiesFragment(FileDetailActivitiesFragment.java:331)
	at com.owncloud.android.ui.fragment.-$$Lambda$FileDetailActivitiesFragment$N7POe6TZvoGcuV2qP-niPxY_7t4.run(Unknown Source:12)
	at java.lang.Thread.run(Thread.java:784)

************ APP INFORMATION ************
ID: com.nextcloud.android.beta
Version: 20200211
Build flavor: versionDev

************ DEVICE INFORMATION ************
Brand: HUAWEI
Device: HWVTR
Model: VTR-L29
Id: HUAWEIVTR-L29
Product: VTR-L29

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: 252C432

**Comment by AndyScherzinger** [comment]
Duplicate of #5331
